crates/ruff_python_formatter/src/expression/expr_constant.rs

```diff
@@ -61,20 +53,32 @@ impl FormatNodeRule<ExprConstant> for FormatExprConstant {
 impl NeedsParentheses for ExprConstant {
     fn needs_parentheses(
         &self,
-        parenthesize: Parenthesize,
+        _parent: AnyNodeRef,
         context: &PyFormatContext,
-    ) -> Parentheses {
-        match default_expression_needs_parentheses(self.into(), parenthesize, context) {
-            Parentheses::Optional if self.value.is_str() && parenthesize.is_if_breaks() => {
-                // Custom handling that only adds parentheses for implicit concatenated strings.
-                if parenthesize.is_if_breaks() {
-                    Parentheses::Custom
-                } else {
-                    Parentheses::Optional
-                }
+    ) -> OptionalParentheses {
+        if self.value.is_str() {
+            let contents = context.locator().slice(self.range());
+            // Don't wrap triple quoted strings
+            if is_multiline_string(self, context.source()) || !is_implicit_concatenation(contents) {
```

Comment:
This is somewhat slow. We should consider extracting the ranges of all multiline strings (similar to Ruff)